src/odemis/gui/cont/tools.py

```diff
@@ -258,8 +276,21 @@ def _add_button(self, cls, img_prefix, tooltip=None):
             f = wx.TOP | wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL
             b = 5
 
-        self.btn_sizer.Add(btn, border=b, flag=f)
-        self.btn_panel.Layout()
+        if self.orientation == (wx.VERTICAL | wx.HORIZONTAL):
+            if self.tool == TOOL_RULER:
+                self.btn_sizer.Add(btn, (0, 0), border=b, flag=wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL)
+            elif self.tool == TOOL_LABEL:
+                self.btn_sizer.Add(btn, (1, 0), border=b, flag=wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL)
+            elif self.tool == TOOL_POINT:
+                self.btn_sizer.Add(btn, (0, 1), border=b, flag=wx.BOTTOM | wx.ALIGN_CENTRE_HORIZONTAL | wx.LEFT)
+            elif self.tool == TOOL_LINE:
```

Comment:
Is the `wx.LEFT` flag really necessary? It'd be simpler it it wasn't needed. If you need margin, you might be able to add it to the sizer itself.